fibers.js

```diff
@@ -0,0 +1,25 @@
+const Future = require('fibers/future');
+const fs = Future.wrap(require('fs'));
+const prependDir = (file) => require('path').join(process.argv[2], file)
+
+const main = () => {
+  Future.task(function() {
+    try {
+      process.stdout.write(
+        fs
+        .readFileFuture(prependDir('index.txt'))
+        .wait()
+        .toString()
+        .replace(/\n$/,'')
+        .split('\n')
+        .map((x) => fs.readFileFuture(prependDir(x)).wait().toString())
+        .join(''))
```

Comment:
Is all of this code impure? The idea of using a Future/Task type is to primarily write pure code, with small bits of effectful code at the very edges of our programs. In __tasks.js__, [`concatFiles`][1] is a pure function. Could we do something similar here? If not, this Future implementation seems broken to me. ;)


[1]: https://github.com/plaid/async-problem/blob/d15c4f8fb8170d733f00557da694616230db1d63/tasks.js#L28-L36